### PR TITLE
Member dropdown

### PR DESCRIPTION
**Pull request by eldiegod**
Fixes: #1645

Testing path: '/playground/member-dropdown'

Figma: https://www.figma.com/file/Pf31tuYpozYmpq163U2ho8/Web-Components?node-id=110%3A18281

Missing things to come later at some point:
- Real Tjoy balance
- Learn about tJOY link
- my profile link
- switch member animation

### CODE REVIEW

File: src/components/_overlays/MemberDropdown/MemberDropdown.tsx
Comment (drillprop) [inline src/components/_overlays/MemberDropdown/MemberDropdown.tsx]: We had already `handleSwitchMemberMode` which we could use here as well(you'd have to add a parameter to the function of course)

File: src/components/_overlays/MemberDropdown/MemberDropdown.tsx
Comment (drillprop) [inline src/components/_overlays/MemberDropdown/MemberDropdown.tsx]:
```suggestion
              label="Add new member..."
```
Ok, we should have some eslint/prettier plugin already for changing that 😅  I think [this one](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-curly-brace-presence.md) should work, but let's do this in another PR.

File: src/components/_overlays/MemberDropdown/MemberDropdown.tsx
Comment (drillprop) [inline src/components/_overlays/MemberDropdown/MemberDropdown.tsx]: Let's collapse this empty tag

File: src/components/_overlays/MemberDropdown/MemberDropdown.tsx
Comment (drillprop) [inline src/components/_overlays/MemberDropdown/MemberDropdown.tsx]: In the design `Learn about tJOY` isn't underlined. 
image
But to be honest I like underlined text here. It's suggesting that it's a clickable link. Could you ask @toiletgranny if we could leave this without change as you have here?

Comment (drillprop) [reply src/components/_overlays/MemberDropdown/MemberDropdown.tsx]: btw. I'm fine with @ts-ignore

File: src/components/_overlays/MemberDropdown/MemberDropdown.tsx
Comment (drillprop) [inline src/components/_overlays/MemberDropdown/MemberDropdown.tsx]: I think we don't need a div here, we could use Fragment.

**Review by drillprop — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/51168865/146333258-022665a1-8bc4-412d-a9f5-656cb24d9f16.mov

I've noticed some layout shifts when switching members. Could we do something about it?
Besides that, found some minor stuff. Looks pretty good in general 👍🏻

**Comment by kdembler** [comment]
Is there any specific reason to not use it in Studio in place of the old one right away?

File: src/components/_overlays/MemberDropdown/MemberDropdown.tsx
Comment (eldiegod) [reply src/components/_overlays/MemberDropdown/MemberDropdown.tsx]: @toiletgranny what do you think?

File: src/components/_overlays/MemberDropdown/MemberDropdown.tsx
Comment (eldiegod) [reply src/components/_overlays/MemberDropdown/MemberDropdown.tsx]: I think since the func is just the setter, it's not needed, will remove and use the setter instead for now

**Comment by eldiegod** [comment]
> Is there any specific reason to not use it in Studio in place of the old one right away?

@kdembler added it to studio topbar but now we are lacking the logout action

File: src/components/_overlays/MemberDropdown/MemberDropdown.tsx
Comment (drillprop) [reply src/components/_overlays/MemberDropdown/MemberDropdown.tsx]: Talked to Adam about this, and he decided that it should be underlined only when you're hovering over this link.

**Comment by toiletgranny** [comment]
@eldiegod, @kdembler, @drillprop Great work, looks stunning! Can't wait to see that transition animation when going into "Switch member" mode! For all other comments and questions, I recorded a quick loom (4 min). https://www.loom.com/share/435e049ab7ce476e8510c2051af73578

**Comment by eldiegod** [comment]
@toiletgranny I think I implemented everything you mentioned besides the `Select first channel when switching members to a member with channels` part because this is probably more suitable as an issue on its own since it has more to do with our internal user logic than with the member dropdown itself

**Comment by drillprop** [comment]
> besides the `Select first channel when switching members to a member with channels` part because this is probably more suitable as an issue on its own since it has more to do with our internal user logic than with the member dropdown itself 

I'll deal with it in https://github.com/Joystream/atlas/pull/1788 or in follow-up PR once this member dropdown PR is merged. I also think that it will be better to merge your commits into `Joystream:signin`, not `Joystream:dev`, wdyt?

File: src/components/_navigation/TopbarStudio/TopbarStudio.tsx
Comment (drillprop) [inline src/components/_navigation/TopbarStudio/TopbarStudio.tsx]: Let's just remove this

File: src/components/ListItem/ListItem.tsx
Comment (drillprop) [inline src/components/ListItem/ListItem.tsx:63]: hmm, that's a cool library!

Comment (eldiegod) [reply src/components/ListItem/ListItem.tsx:63]: yea right

**Comment by toiletgranny** [comment]
> @toiletgranny I think I implemented everything you mentioned besides the `Select first channel when switching members to a member with channels` part because this is probably more suitable as an issue on its own since it has more to do with our internal user logic than with the member dropdown itself

Ok, thanks! Visually, everything looks good now! Based on your reply, I'd assume that closing the member dropdown upon selecting a different member, which I also mentioned in the loom video, would also be tackled in that separate issue, right?
CleanShot 2021-12-20 at 12 25 55

Other than this — it's a go from my side!

**Comment by eldiegod** [comment]
> > besides the `Select first channel when switching members to a member with channels` part because this is probably more suitable as an issue on its own since it has more to do with our internal user logic than with the member dropdown itself
> 
> I'll deal with it in #1788 or in follow-up PR once this member dropdown PR is merged. I also think that it will be better to merge your commits into `Joystream:signin`, not `Joystream:dev`, wdyt?

Nice!

~~About merging; I'm not sure, I would merge to dev to not make that release bigger but is not clear, thoughts @kdembler ?~~

nvm we are merging to signin

**Comment by eldiegod** [comment]
> > @toiletgranny I think I implemented everything you mentioned besides the `Select first channel when switching members to a member with channels` part because this is probably more suitable as an issue on its own since it has more to do with our internal user logic than with the member dropdown itself
> 
> Ok, thanks! Visually, everything looks good now! Based on your reply, I'd assume that closing the member dropdown upon selecting a different member, which I also mentioned in the loom video, would also be tackled in that separate issue, right? CleanShot 2021-12-20 at 12 25 55
> 
> Other than this — it's a go from my side!

@toiletgranny  No we don't have to wait for that I just added it so we should be feature complete here

File: src/components/_overlays/MemberDropdown/MemberDropdown.styles.ts
Comment (kdembler) [inline src/components/_overlays/MemberDropdown/MemberDropdown.styles.ts]: Couldn't we use just `100vh` instead? We have a guarantee this will always be responsive, `window.innerHeight` is iffy

**Review by kdembler — COMMENTED** [review]
Just a small comment from myself, other than that looks great to me!

I will still wait for final approval from @drillprop

**Review by drillprop — APPROVED** [review]
Looks good to me! 🎉

File: src/components/_overlays/MemberDropdown/MemberDropdown.styles.ts
Comment (eldiegod) [reply src/components/_overlays/MemberDropdown/MemberDropdown.styles.ts]: yeah this is better just pushed it

**Review by kdembler — APPROVED** [review]

**Pull request merged**
